crates/revm/src/batch.rs

```diff
@@ -84,7 +84,7 @@ impl BlockBatchRecord {
     }
 
     /// Returns all recorded requests.
-    pub fn take_requests(&mut self) -> Requests {
+    pub fn take_requests(&mut self) -> Vec<Requests> {
```

Comment:
I think we start having a lot of these `Vec<Vec<_>>` outer / inner vecs so maybe we should do a generic `Batch<T>` to encapsulate that